inference-engine/src/mkldnn_plugin/nodes/mkldnn_color_convert_node.cpp

```diff
@@ -0,0 +1,666 @@
+                dst + (batch * width * height + h * width) * 3,
+                width,
+                _colorFormat[0]     // The first byte is enough to determine the RGB or BGR format.
+            };
+
+            kernel(args);
+        });
+    }
+};
+
+template<typename T>
+class TwoPlaneConvert<T, impl_desc_type::jit_uni> : public Converter {
+public:
+    using Converter::Converter;
+
+    void execute(mkldnn::stream strm) override {
+        const auto & kernel = jit_uni_converter::get<T>();
+        const auto & dims = inputDims(0);
+
+        const size_t batch_size = dims[N_DIM];
+        const size_t height = dims[H_DIM];
+        const size_t width = dims[W_DIM];
+
+        const T* y = static_cast<const T*>(input(0));
+        const T* uv = static_cast<const T*>(input(1));
+        T* dst = static_cast<T*>(output(0));
+
+        const size_t stride_y = height * width;
+        const size_t stride_uv = height * width / 2;
+
+        InferenceEngine::parallel_for2d(batch_size, height, [&](int batch, int h) {
+            typename jit_uni_converter::Params args = {
+                y + batch * stride_y + h * width,
+                uv + batch * stride_uv + (h / 2) * width,
+                dst + (batch * width * height + h * width) * 3,
+                width,
+                _colorFormat[0]     // The first byte is enough to determine the RGB or BGR format.
+            };
+
+            kernel(args);
+        });
+    }
+};
+
+}   // namespace nv12
+}   // namespace
+
+MKLDNNColorConvertNode::Converter::Converter(MKLDNNNode *node, const ColorFormat & colorFormat)
+    : _node(node)
+    , _colorFormat(colorFormat) {
+}
+
+InferenceEngine::Precision MKLDNNColorConvertNode::Converter::inputPrecision(size_t idx) const {
+    return _node->getParentEdgeAt(idx)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getPrecision();
+}
+
+InferenceEngine::Precision MKLDNNColorConvertNode::Converter::outputPrecision(size_t idx) const {
+    return _node->getChildEdgeAt(idx)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getPrecision();
+}
+
+const void * MKLDNNColorConvertNode::Converter::input(size_t idx) const {
+    return _node->getParentEdgeAt(idx)->getMemoryPtr()->GetPtr();
+}
+
+void * MKLDNNColorConvertNode::Converter::output(size_t idx) const {
+    return _node->getChildEdgeAt(idx)->getMemoryPtr()->GetPtr();
+}
+
+const VectorDims & MKLDNNColorConvertNode::Converter::inputDims(size_t idx) const {
+    return _node->getParentEdgesAtPort(idx)[0]->getMemory().getStaticDims();
+}
+
+MKLDNNColorConvertNode::MKLDNNColorConvertNode(const std::shared_ptr<ngraph::Node>& op,
+                                               const mkldnn::engine& eng,
+                                               MKLDNNWeightsSharing::Ptr &cache)
+    : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    std::tie(algorithm, errorMessage) = getAlgorithmFor(op);
+    if (algorithm == Algorithm::Default)
```

Comment:
Isn't it better to have those checks in form of isSupportedOperation call, again, to align with other nodes?

Reply:
The isSupportedOperation sounds like it should check only the supported operation but I need determine algorithm as well. I don't want to check nGraph node types twice. ):

Reply:
I mean all other nodes are basically doing the same (some kind of double work)
From my understanding isSupportedOperation is kind of "must" function for every node.
I propose to leave this comment for final review from @dmitry-gorokhov

Reply:
isSupportedOperation added